Projects/Scripts/Misc/Email.cs

```diff
@@ -1,98 +1,143 @@
+using MailKit.Net.Smtp;
+using MimeKit;
+using Server.Accounting;
+using Server.Engines.Help;
 using System;
-using System.Net;
-using System.Net.Mail;
+using System.IO;
 using System.Text.RegularExpressions;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Server.Misc
 {
   public class Email
   {
-    /* In order to support emailing, fill in EmailServer and FromAddress:
-     * Example:
-     *  public static readonly string EmailServer = "mail.domain.com";
-     *  public static readonly string FromAddress = "runuo@domain.com";
-     * 
-     * If you want to add crash reporting emailing, fill in CrashAddresses:
-     * Example:
-     *  public static readonly string CrashAddresses = "first@email.here,second@email.here,third@email.here";
-     * 
-     * If you want to add speech log page emailing, fill in SpeechLogPageAddresses:
-     * Example:
-     *  public static readonly string SpeechLogPageAddresses = "first@email.here,second@email.here,third@email.here";
-     */
-
-    public static readonly string EmailServer = null;
-    public static readonly int EmailPort = 25;
-
-    public static readonly string FromAddress = null;
-    public static readonly string EmailUsername = null;
-    public static readonly string EmailPassword = null;
+    public static readonly MailboxAddress FromAddress = new MailboxAddress(Configuration.Instance.emailSettings.FromName, Configuration.Instance.emailSettings.FromAddress);
+    public static readonly MailboxAddress CrashAddresses = new MailboxAddress(Configuration.Instance.emailSettings.crashName, Configuration.Instance.emailSettings.crashAddress);
+    public static readonly MailboxAddress SpeechLogPageAddress = new MailboxAddress(Configuration.Instance.emailSettings.speechLogPageName, Configuration.Instance.emailSettings.speechLogPageAddress);
+    public static readonly string EmailServer = Configuration.Instance.emailSettings.emailServer;
+    public static readonly int EmailPort = Configuration.Instance.emailSettings.emailPort;
+    public static readonly string EmailServerUsername = Configuration.Instance.emailSettings.emailUsername;
+    public static readonly string EmailServerPassword = Configuration.Instance.emailSettings.emailPassword;
+    public static readonly int m_retryCount = 5;

```

Comment:
Maybe this should also be a configuration?

Also we should come up with a standard for public static variable naming. I am up for suggestions. Sometimes SCREAMING_UNDERSCORE is used.

Reply:
So EMAIL_PORT, EMAIL_SERVER_USERNAME ?